sdk/azidentity/default_azure_credential.go

```diff
@@ -57,7 +58,14 @@ func NewDefaultAzureCredential(options *DefaultAzureCredentialOptions) (*Default
 		creds = append(creds, &defaultCredentialErrorReporter{credType: "EnvironmentCredential", err: err})
 	}
 
-	msiCred, err := NewManagedIdentityCredential(&ManagedIdentityCredentialOptions{ClientOptions: options.ClientOptions})
+	var managedClientID ClientID
+	if s := os.Getenv("AZURE_CLIENT_ID"); s != "" {
+		managedClientID = ClientID(s)
+	}
+	msiCred, err := NewManagedIdentityCredential(&ManagedIdentityCredentialOptions{
+		ClientOptions: options.ClientOptions,
+		ID:            managedClientID,
+	})
```

Comment:
When `ManagedIdentityCredentialOptions.ID.String() == ""`, the client adds an empty query parameter to token requests. Better to leave it `nil` when AZURE_CLIENT_ID isn't set:
```suggestion
	o := &ManagedIdentityCredentialOptions{ClientOptions: options.ClientOptions}
	if ID, ok := os.LookupEnv("AZURE_CLIENT_ID"); ok {
		o.ID = ClientID(ID)
	}
	msiCred, err := NewManagedIdentityCredential(o)
```

Reply:
Fixed as suggested.
Thanks.